tests/apollo/test_skvbc_view_change.py

```diff
@@ -394,6 +394,68 @@ async def test_single_vc_current_and_next_primaries_down(self, bft_network, trac
             num_consecutive_failing_primaries = 2
         )
 
+    @unittest.skipIf(environ.get('BUILD_COMM_TCP_TLS', "").lower() == "true", "Unstable on CI (TCP/TLS only)")
+    @with_trio
+    @with_bft_network(start_replica_cmd, rotate_keys=True)
+    async def test_replica_asks_to_leave_view(self, bft_network):
+        """
+        This test makes sure that a single isolated replica will ask for a view change, but it doesn't occur:
+        1) Start all replicas
+        2) Isolate a single replica for long enough to trigger a ReplicaAsksToLeaveViewMsg
+        3) Let the replica rejoin the consensus
+        4) Make sure a view change does not happen and the isolated replica
+        rejoins the fast path in the existing view
+        """
+        
+        bft_network.start_all_replicas()
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network)
+        n = bft_network.config.n
+
+        self.assertEqual(len(bft_network.procs), n,
+                         "Make sure all replicas are up initially.")
+
+        initial_primary = await bft_network.get_current_primary()
+
+        isolated_node = random.choice(
+            bft_network.all_replicas(without={initial_primary}))
+        isolated_replicas = set([isolated_node])
+
+        with net.ReplicaSubsetTwoWayIsolatingAdversary(bft_network, isolated_replicas) as adversary:
```

Comment:
Right before activating the adversary, let's first assert that `sentReplicaAsksToLeaveViewMsgDueToStatus` is `0`.

Reply:
@KTeofanidis what about this one? How can we check that the replica does not complain when the full BFT network is up and running?

Thanks!
-- Teo